packages/webapp/pages/onboarding.tsx

```diff
@@ -209,126 +236,46 @@ export function OnboardPage(): ReactElement {
     setActiveScreen(OnboardingStep.EditTag);
   };
 
-  useEffect(() => {
-    if (!isPageReady || isLogged.current) {
-      return;
-    }
-
-    if (user) {
-      router.replace(getPathnameWithQuery(webappUrl, window.location.search));
-      return;
-    }
-
-    isLogged.current = true;
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [logEvent, isPageReady, user]);
-
-  useEffect(() => {
-    setHasSelectTopics(!!feedSettings?.includeTags?.length);
-  }, [feedSettings?.includeTags?.length]);
-
-  const getAuthOptions = () => {
-    return (
-      <AuthOptions
-        simplified
-        className={{
-          container: classNames(
-            'w-full rounded-none',
-            maxAuthWidth,
-            isAuthenticating && 'h-full',
-            !isAuthenticating && 'max-w-full',
-          ),
-          onboardingSignup: '!gap-5 !pb-5 tablet:gap-8 tablet:pb-8',
-        }}
-        trigger={AuthTriggers.Onboarding}
-        formRef={formRef}
-        defaultDisplay={defaultDisplay}
-        forceDefaultDisplay={!isAuthenticating}
-        initialEmail={email}
-        isLoginFlow={isLoginFlow}
-        targetId={targetId}
-        onSuccessfulLogin={onSuccessfulLogin}
-        onSuccessfulRegistration={onSuccessfulRegistration}
-        onAuthStateUpdate={(props: AuthProps) =>
-          setAuth({ isAuthenticating: true, ...props })
-        }
-        onboardingSignupButton={{
-          size: isMobile ? ButtonSize.Medium : ButtonSize.Large,
-          variant: ButtonVariant.Primary,
-        }}
-      />
-    );
-  };
+  const authOptionProps: AuthOptionsProps = useMemo(() => {
```

Comment:
See discussion:
https://dailydotdev.slack.com/archives/C02E2C3C13R/p1729060974358809

This was the best middleground I think :)